cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/builder/Fluent.kt

```diff
@@ -0,0 +1,385 @@
+): CallExpression {
+    // Try to parse the name
+    val parsedName = parseName(name)
+    val node =
+        if (parsedName.parent != null) {
+            newMemberCallExpression(
+                newMemberExpression(
+                    parsedName.localName,
+                    newDeclaredReferenceExpression(parsedName.parent, parseType(parsedName.parent))
+                ),
+                isStatic
+            )
+        } else {
+            newCallExpression(newDeclaredReferenceExpression(parsedName))
+        }
+    if (init != null) {
+        init(node)
+    }
+
+    val holder = this@Holder
+    if (holder is StatementHolder) {
+        holder += node
+    } else if (holder is ArgumentHolder) {
+        holder += node
+    }
+
+    return node
+}
+
+/**
+ * Creates a new [IfStatement] in the Fluent Node DSL and adds it to the
+ * [StatementHolder.statements] of the nearest enclosing [StatementHolder]. The [init] block can be
+ * used to create further sub-nodes as well as configuring the created node itself.
+ */
+context(StatementHolder)
+
+fun LanguageFrontend.ifStmt(init: IfStatement.() -> Unit): IfStatement {
+    val node = newIfStatement()
+    init(node)
+
+    (this@StatementHolder) += node
+
+    return node
+}
+
+/**
+ * Configures the [IfStatement.condition] in the Fluent Node DSL of the nearest enclosing
+ * [IfStatement]. The [init] block can be used to create further sub-nodes as well as configuring
+ * the created node itself.
+ */
+fun IfStatement.condition(init: IfStatement.() -> BinaryOperator): BinaryOperator {
+    return init(this)
+}
+
+/**
+ * Creates a new [CompoundStatement] in the Fluent Node DSL and sets it to the
+ * [IfStatement.thenStatement] of the nearest enclosing [IfStatement]. The [init] block can be used
+ * to create further sub-nodes as well as configuring the created node itself.
+ */
+context(IfStatement)
+
+fun LanguageFrontend.then(
+    needsScope: Boolean = true,
+    init: CompoundStatement.() -> Unit
+): CompoundStatement {
+    val node = newCompoundStatement()
+    scopeIfNecessary(needsScope, node, init)
+
+    thenStatement = node
+
+    return node
+}
+
+/**
+ * Creates a new [IfStatement] in the Fluent Node DSL and sets it to the [IfStatement.elseStatement]
+ * of the nearest enclosing [IfStatement]. This simulates an `else-if` scenario. The [init] block
+ * can be used to create further sub-nodes as well as configuring the created node itself.
+ */
+context(IfStatement)
+
+fun LanguageFrontend.elseIf(init: IfStatement.() -> Unit): IfStatement {
+    val node = newIfStatement()
+    init(node)
+
+    elseStatement = node
+
+    return node
+}
+
+/**
+ * Creates a new [CompoundStatement] in the Fluent Node DSL and sets it to the
+ * [IfStatement.elseStatement] of the nearest enclosing [IfStatement]. The [init] block can be used
+ * to create further sub-nodes as well as configuring the created node itself.
+ */
+context(IfStatement)
+
+fun LanguageFrontend.els(
```

Comment:
The name `els` is kind of weird but I'm not sure what would be better. `elseStmt` is more long but would probably be less confusing

Reply:
Should I also rename `then` to `thenStmt`?

Reply:
For consistency reasons it could make sense but to me, it would be ok like this as well

Reply:
Done